File: codes/week17/진현/121.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0, min_price = INT_MAX;
    
        for (int i=0; i<prices.size(); i++) {
            min_price = min(prices[i], min_price);

            if (prices[i] - min_price > profit)
                profit = prices[i] - min_price;
        }

        return profit;
    }
};

File: codes/week17/지원/121.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        answer = 0
        min_price = float('inf')  
        max_price = -float('inf')

        for price in prices:
            if price < min_price:
                min_price = price
                max_price = price
         
            if price > max_price:
                max_price = price
            
            answer = max(answer, max_price - min_price)

        return answer